src/promptflow/tests/sdk_cli_test/unittests/test_utils.py

```diff
@@ -199,6 +202,19 @@ def test_concurrent_execution_of_refresh_connections_dir(self, concurrent_count)
         for thread in threads:
             thread.join()
 
+    @pytest.mark.parametrize("concurrent_count", [1, 2, 4, 8])
+    def test_concurrent_hint_for_update(self, concurrent_count):
+        from concurrent.futures import ThreadPoolExecutor
+        with patch('promptflow._utils.version_hint_utils.datetime') as mock_datetime:
+            mock_datetime.datetime.now.return_value = datetime.datetime.now()
+            mock_datetime.datetime.strptime.return_value = datetime.datetime.now() - datetime.timedelta(days=8)
+            mock_datetime.timedelta.return_value = datetime.timedelta(days=7)
+            hint_for_update()
+            with ThreadPoolExecutor(max_workers=concurrent_count) as pool:
+                pool.submit(check_latest_version)
+
```

Comment:
Will there be cases check_latest_version not finished when running the following line?

Reply:
No, we won't check since Clement suggests may not break user because of get pypi.